assert/utils.go

```diff
@@ -1350,6 +1350,58 @@ func formatEndsWithError(actual string, expected string, actualEndSufix string,
 	return ""
 }
 
+// findExactCaseMismatch finds the exact case mismatch for a substring within a string
+// Returns the position and the found substring if there's an exact case-only difference
+// If actualLower and substringLower are provided, they will be used to avoid recomputing
+func findExactCaseMismatch(actual, substring string, precomputedLower ...string) (int, string, bool) {
+	if len(substring) == 0 {
+		return -1, "", false
+	}
+
+	var actualLower, substringLower string
+	if len(precomputedLower) >= 2 {
+		// Use precomputed lowercase versions for better performance
+		actualLower = precomputedLower[0]
+		substringLower = precomputedLower[1]
+	} else {
+		actualLower = strings.ToLower(actual)
+		substringLower = strings.ToLower(substring)
+	}
+
+	// Check if there's a case-insensitive match
+	index := strings.Index(actualLower, substringLower)
+	if index == -1 {
+		return -1, "", false
+	}
+
+	// Extract the actual substring from the original text
+	foundSubstring := actual[index : index+len(substring)]
+
+	// Check if it's exactly the same except for case
+	if strings.EqualFold(foundSubstring, substring) && foundSubstring != substring {
+		return index, foundSubstring, true
+	}
+
+	return -1, "", false
+}
+
+// formatSimpleCaseMismatchError formats a simplified error message for exact case mismatches
+func formatSimpleCaseMismatchError(substring, foundSubstring string, position int) string {
+	// Clean empty strings for display
+	displayNeedle := substring
+	displayFound := foundSubstring
+
+	var b strings.Builder
+	fmt.Fprintf(&b,
+		"Expected string to contain '%s', but found case difference\n"+
+			"Substring: '%s'\n"+
```

Comment:
Be careful on how a string with a quote.

Using %q is often preferred 


```suggestion
			"Substring: %q\n"+
```

But you could also use something about quote detection with strconv.CanBackQuotes in a formatter, but here I feel my comment is out of the scope of this PR